starlite/contrib/open_telemetry/middleware.py

```diff
@@ -0,0 +1,51 @@
+from typing import TYPE_CHECKING
+
+from starlite.exceptions import MissingDependencyException
+from starlite.middleware.base import AbstractMiddleware
+
+try:
+    from opentelemetry.instrumentation.asgi import OpenTelemetryMiddleware
+except ImportError as e:
+    raise MissingDependencyException("OpenTelemetry dependencies are not installed") from e
+
+if TYPE_CHECKING:
+    from starlite.contrib.open_telemetry import OpenTelemetryConfig
+    from starlite.types import ASGIApp, Receive, Scope, Send
+
+
+class OpenTelemetryInstrumentationMiddleware(AbstractMiddleware):
+    """OpenTelemetry Middleware."""
+
+    __slots__ = ("open_telemetry_middleware",)
+
+    def __init__(self, app: "ASGIApp", config: "OpenTelemetryConfig"):
+        """Middleware that adds OpenTelemetry instrumentation to the application.
+
+        Args:
+            app: The 'next' ASGI app to call.
+            config: An instance of [OpenTelemetryConfig][starlite.contrib.open_telemetry.OpenTelemetryConfig]
+        """
+        super().__init__(app=app, scopes=config.scopes, exclude=config.exclude, exclude_opt_key=config.exclude_opt_key)
```

Comment:
OpenTelemetry also provides the ability to exclude specific routes with `OTEL_PYTHON_{LIB}_EXCLUDED_URLS` with a comma delimited string. It's probably worth supporting it?

Reply:
we also have a mechanism for this, but not via env variables - rather using the config. I thus decided not to incldude this. But perhaps its better that we do, so I'll add this.